k2/csrc/fsa.h

```diff
@@ -13,7 +13,7 @@
 namespace k2 {
 
 using Label = int32_t;
-using StateId = int32_t;
+using StateId = std::size_t;
```

Comment:
Is it really necessary to use unsigned `std::size_t` here?

In OpenFST, `-1` indicates an invalid state id.

Comparison between `int` and `std::size_t` may result in warnings later.